addons/account/tests/common.py

```diff
@@ -985,6 +832,169 @@ def setup_armageddon_tax(cls, tax_name, company_data):
+                    'account_dest_id': obj.product_b.property_account_expense_id.id,
+                }),
+            ],
+        })
+
+        # ==== Payment terms ====
+        obj.pay_terms_a = obj.env.ref('account.account_payment_term_immediate')
+        obj.pay_terms_b = obj.env['account.payment.term'].create({
+            'name': '30% Advance End of Following Month',
+            'note': 'Payment terms: 30% Advance End of Following Month',
+            'line_ids': [
+                (0, 0, {
+                    'value': 'percent',
+                    'value_amount': 30.0,
+                    'sequence': 400,
+                    'days': 0,
+                    'option': 'day_after_invoice_date',
+                }),
+                (0, 0, {
+                    'value': 'balance',
+                    'value_amount': 0.0,
+                    'sequence': 500,
+                    'days': 31,
+                    'option': 'day_following_month',
+                }),
+            ],
+        })
+
+        # ==== Partners ====
+        obj.partner_a = obj.env['res.partner'].create({
+            'name': 'partner_a',
+            'property_payment_term_id': obj.pay_terms_a.id,
+            'property_supplier_payment_term_id': obj.pay_terms_a.id,
+            'property_account_receivable_id': obj.company_data['default_account_receivable'].id,
+            'property_account_payable_id': obj.company_data['default_account_payable'].id,
+            'company_id': False,
+        })
+        obj.partner_b = obj.env['res.partner'].create({
+            'name': 'partner_b',
+            'property_payment_term_id': obj.pay_terms_b.id,
+            'property_supplier_payment_term_id': obj.pay_terms_b.id,
+            'property_account_position_id': obj.fiscal_pos_a.id,
+            'property_account_receivable_id': obj.company_data['default_account_receivable'].copy().id,
+            'property_account_payable_id': obj.company_data['default_account_payable'].copy().id,
+            'company_id': False,
+        })
+
+        # ==== Cash rounding ====
+        obj.cash_rounding_a = obj.env['account.cash.rounding'].create({
+            'name': 'add_invoice_line',
+            'rounding': 0.05,
+            'strategy': 'add_invoice_line',
+            'profit_account_id': obj.company_data['default_account_revenue'].copy().id,
+            'loss_account_id': obj.company_data['default_account_expense'].copy().id,
+            'rounding_method': 'UP',
+        })
+        obj.cash_rounding_b = obj.env['account.cash.rounding'].create({
+            'name': 'biggest_tax',
+            'rounding': 0.05,
+            'strategy': 'biggest_tax',
+            'rounding_method': 'DOWN',
+        })
+
+    # -------------------------------------------------------------------------
+    # HELPERS
+    # -------------------------------------------------------------------------
+
+    @classmethod
+    def copy_account(cls, account):
```

Comment:
I know it's not sexy but I want to keep the `cls.copy_account` instead of importing every time `from common import AccountTestSetup` and then `AccountTestSetup.copy_account(account)` Otherwise you are right, I think it doesn't work for `AccountTestInvoicingHttpCommon` in that case...